rskj-core/src/main/java/org/ethereum/rpc/exception/RskErrorResolver.java

```diff
@@ -21,16 +22,32 @@
 
     @Override
     public JsonError resolveError(Throwable t, Method method, List<JsonNode> arguments) {
-        JsonError error = null;
-        if(t instanceof  RskJsonRpcRequestException) {
-            error =  new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
+        JsonError error;
+
+        if (t instanceof RskJsonRpcRequestException) {
+            error = new JsonError(((RskJsonRpcRequestException) t).getCode(), t.getMessage(), null);
         } else if (t instanceof InvalidFormatException) {
             error = new JsonError(JsonRpcError.INTERNAL_ERROR, "Internal server error, probably due to invalid parameter type", null);
         } else if (t instanceof UnrecognizedPropertyException) {
             error = new JsonError(
                     JsonRpcError.INVALID_PARAMS,
                     getExceptionMessage((UnrecognizedPropertyException) t),
                     null);
+        } else if (t.getMessage() != null && t.getMessage().contains("An RSK address must be")) {
```

Comment:
Identifying errors from strings isn't usually regarded as best practice, couldn't we use `getCode()` instead?

Reply:
The thing is that, this is a `RuntimeException` so it is really difficult to know from where it was thrown. This one is thrown from `RskAddress.java`

Reply:
Do we know where these errors are thrown? Maybe it is worth it to make specific exceptions for them... otherwise, LGTM.

Reply:
I remember creating it for this commit: https://github.com/rsksmart/rskj/pull/1669/commits/9d40cfeab15196dbf89ddb88547e96ba0e86d47a

but for some it was recommended not to do that, but since has been a lot days I don't clearly remember.

maybe @Vovchyk could remember why we are not creating a exception for this case

Reply:
> maybe @Vovchyk could remember why we are not creating a exception for this case

not sure what case you mean here, but I agree with @joaquinlpereyra. Better to have custom exceptions / error codes and not rely on strings, which tend to change

Reply:
@Vovchyk You can take a look to the commit I mentioned, where I have created a class for it and then it was deleted, Is it save to revert that class back? I remembered to remove it for some reason. But if you agree and can add that change specifically.

Reply:
@rmoreliovlabs what class are you talking about? :)

Reply:
ok, do you mean this [comment](https://github.com/rsksmart/rskj/pull/1669#discussion_r770330403)? ) the comment was related to the fact that the class wasn't in use, wasn't it?

Reply:
Yeah it was in use, I created in order to make it easier to know when I get this to precisely avoid using strings to check the error type.

I will add the change again

Reply:
Done

Reply:
> RskJsonRpcRequestException is an JSON-RPC layer error, but RskAddress is a core class that's being used in multiple layers. I wouldn't introduce this kind of exception in this class

ok, I guess I understand what you meant now. In this [comment](https://github.com/rsksmart/rskj/pull/1669#discussion_r767905522) I meant that `RpcInvalidRskAddressException` is an JSON-RPC layer error, at least it contains the `Rpc` prefix and extends `RskJsonRpcRequestException`. `RskAddress` is a core class that's being used in multiple layers, not only at the json-rpc level